### PR TITLE
Refactor/400 use custom swr hooks

### LINKED ISSUE #400: Refactor expensive queries by using hooks
We have a hooks folder, we better use custom SWR hooks when performing complicated back end calls so we always use the SWR config we want.

### PR DESCRIPTION
**Pull request by jitsedesmet**
This Pr aims to close #400.

### CODE REVIEW

File: frontend/src/components/project_assignment/assignmentItem.tsx [hunk 1]
```diff
@@ -1,94 +1,122 @@
...
  * Furthermore, it contains a delete button to delete the assignment.
- * @param item Skill you want the assignments from
+ * @param skill [IProjectSkill] you want the assignments from
  * @constructor
  */
-function AssignmentItem(item: { skill: IProjectSkill }) {
+function AssignmentItem({ skill }: IAssignmentItemProps) {
     const { t } = useTranslation("common");
+    const { mutate } = useSWRConfig();
+    const { data: resAssignments, error: assignmentsError } = useSWR(
```
Comment (andpuydt) [inline frontend/src/components/project_assignment/assignmentItem.tsx:65]: What does "resAssignments" stand for?

File: frontend/src/components/project_assignment/assignmentStudentRow.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+export default function AssignmentStudentRow({
+    assignment,
+    removeCallback,
+    registerSortKey,
+}: IAssignmentStudentProps) {
+    const { t } = useTranslation("common");
+    const { data: resStudent, error: studentError } = useSWR(assignment._links.student.href, getStudentOnUrl);
+    const { data: resAssigner, error: assignerError } = useSWR(assignment._links.assigner.href, getUserOnUrl);
```
Comment (andpuydt) [inline frontend/src/components/project_assignment/assignmentStudentRow.tsx:33]: Where does the "res" come from?

File: frontend/src/pages/projects/create.tsx [hunk 3]
```diff
@@ -5,6 +5,8 @@ import styles from "../../styles/Home.module.css";
 import { CreateProjectForm } from "../../components/createProjectForm";
 import { createProjectSubmitHandler } from "../../handlers/createProjectSubmitHandler";
 import NavBar from "../../components/navBar";
+import { useEditionAPIUrlTransformer } from "../../hooks/utilHooks";
+import { useSWRConfig } from "swr";
```
Comment (andpuydt) [inline frontend/src/pages/projects/create.tsx:9]: Why did you add these imports?

File: frontend/test/integration/Provide.tsx [hunk 4]
```diff
@@ -11,6 +11,11 @@ import { IUser } from "../../src/api/entities/UserEntity";
 
 jest.mock("next/router", () => require("next-router-mock"));
 
+/**
+ * Function that makes sure teh SWR cache is disabled in the provided component.
```
Comment (andpuydt) [inline frontend/test/integration/Provide.tsx:15]:
```suggestion
 * Function that makes sure the SWR cache is disabled in the provided component.
```

**Review by andpuydt — COMMENTED** [review]
Looks good! Can you just alter the directory name "project_assignement" to "projectAssignment" for consistency in the naming?

File: frontend/src/components/project_assignment/assignmentModal.tsx [hunk 5]
```diff
@@ -41,10 +44,7 @@ function AssignmentModal(props: ModalProps) {
                     {capitalize(t("suggesting"))}
                     {props.studentName} {capitalize(t("to project"))}
                     <i>{props.projectName}</i> {capitalize(t("for role"))}
-                    <Badge bg="" style={{ backgroundColor: props.skillColor }}>
-                        {props.skillName}
-                    </Badge>
-                    .
+                    <SkillBadge skill={props.skillName} />.
```
[the remark this answers is not in the document]
Comment (jitsedesmet) [reply frontend/src/components/project_assignment/assignmentModal.tsx:47]: No clue. Looking at the dot it was there already?

File: frontend/src/components/project_assignment/assignmentItem.tsx (hunk 1 above)
Comment (jitsedesmet) [reply frontend/src/components/project_assignment/assignmentItem.tsx:65]: recievedAssignment. Do you have a better suggestion?

File: frontend/src/components/project_assignment/assignmentStudentRow.tsx (hunk 2 above)
Comment (jitsedesmet) [reply frontend/src/components/project_assignment/assignmentStudentRow.tsx:33]: recieved

File: frontend/src/components/project_assignment/assignmentItem.tsx (hunk 1 above)
Comment (andpuydt) [reply frontend/src/components/project_assignment/assignmentItem.tsx:65]: receivedAssignments sounds good to me

File: frontend/src/pages/projects/create.tsx (hunk 3 above)
Comment (jitsedesmet) [reply frontend/src/pages/projects/create.tsx:9]: Oh, seems like they are not needed indeed. Tnx

File: frontend/src/components/project_assignment/assignmentModal.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (andpuydt) [reply frontend/src/components/project_assignment/assignmentModal.tsx:47]: Yes it was, but I'm still wondering why 😅

[the remark this answers is not in the document]
Comment (jitsedesmet) [reply frontend/src/components/project_assignment/assignmentModal.tsx:47]: No clue... Not my code? :stuck_out_tongue:

**Review by andpuydt — APPROVED** [review]
